plugin/trace/setup.go

```diff
@@ -41,7 +41,8 @@ func traceParse(c *caddy.Controller) (*trace, error) {
 	)
 
 	cfg := dnsserver.GetConfig(c)
-	tr.ServiceEndpoint = cfg.HostAddresses()
+	tr.serviceEndpoint = cfg.ListenHosts[0] + ":" + cfg.Port
+
```

Comment:
Here you will have incomplete information if BIND is used with several IP.
However I guess that you will not be able to reuse that IP on another ServerBloc .. so it is still an identifier.